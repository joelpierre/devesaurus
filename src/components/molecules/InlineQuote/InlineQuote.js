import React from 'react';

function InlineQuote({ module, pageTheme }) {
  if (module.__typename !== 'inline_quote') {
    return null;
  }

  return (
    <section className={`primary-main__section theme--${module.theme} inline-quote`}>
      <div className="container-fluid">
        <div className="row">
          <div className="flex">
            <blockquote>
              This is a blockquote
            </blockquote>
          </div>
        </div>
      </div>
    </section>
  );
}

export default InlineQuote;
