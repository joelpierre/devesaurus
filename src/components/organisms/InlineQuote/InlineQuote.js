import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Section from '../../core/Section/Section';
import Row from '../../core/Row/Row';
import Container from '../../core/Container/Container';
import Flex from '../../core/Flex/Flex';
import styles from '../CtaBanner/CtaBanner.module.scss';

const InlineQuote = ({ module, pageTheme }) => {
  return (
    <Section
      data-test="component-inline-quote"
      classes={classNames([
        `theme--${module.theme ? module.theme : pageTheme}`,
        styles['inline-quote'],
      ])}
    >
      <Container>
        <Row>
          <Flex classes="flex">
            <blockquote>This is a blockquote</blockquote>
          </Flex>
        </Row>
      </Container>
    </Section>
  );
};

InlineQuote.propTypes = {
  module: PropTypes.instanceOf(Object).isRequired,
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'tint-alpha',
    'tint-beta',
    'tint-psi',
    'tint-omega',
  ]).isRequired,
};

export default InlineQuote;
