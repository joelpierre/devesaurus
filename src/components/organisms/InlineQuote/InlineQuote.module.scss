@import "~sass/core";

.inline-quote {
  $root: &;

  position: relative;
  width: 100%;
}
