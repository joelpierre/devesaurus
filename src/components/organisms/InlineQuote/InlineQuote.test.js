import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import InlineQuote from './InlineQuote';

const defaultProps = {
  module: {},
  pageTheme: 'alpha',
};

/**
 * Factory function to create a ShallowWrapper for the InlineQuote component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<InlineQuote {...setupProps} />);
};

describe('<InlineQuote/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the InlineQuote Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-inline-quote');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(InlineQuote, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should render with pageTheme instead of module theme', () => {
    wrapper = setup({
      module: {},
      pageTheme: 'brand',
    });
    const component = findByTestAttr(wrapper, 'component-inline-quote');
    matchSnapshot(wrapper);
    expect(component.prop('classes')).toContain('theme--brand');
  });
});
