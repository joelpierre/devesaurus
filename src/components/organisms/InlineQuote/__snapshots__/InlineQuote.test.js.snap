// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineQuote/> should match snapshot 1`] = `
<Section
  classes="theme--alpha inline-quote"
  contrast={false}
  data-test="component-inline-quote"
  type="section"
>
  <Container
    fluid={false}
  >
    <Row
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        classes="flex"
        col={null}
        colLg={null}
        colMd={null}
        colSm={null}
        colXl={null}
        colXs={null}
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <blockquote>
          This is a blockquote
        </blockquote>
      </Flex>
    </Row>
  </Container>
</Section>
`;

exports[`<InlineQuote/> should render with pageTheme instead of module theme 1`] = `
<Section
  classes="theme--brand inline-quote"
  contrast={false}
  data-test="component-inline-quote"
  type="section"
>
  <Container
    fluid={false}
  >
    <Row
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        classes="flex"
        col={null}
        colLg={null}
        colMd={null}
        colSm={null}
        colXl={null}
        colXs={null}
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <blockquote>
          This is a blockquote
        </blockquote>
      </Flex>
    </Row>
  </Container>
</Section>
`;
